### PR TITLE
Update docs for the simulation state dump/load methods

### PR DESCRIPTION
**Pull request by kkg4theweb**
Update docs for the simulation state dump/load methods to cover the new methods added to support field dump/load.

This update also picks up other changes to the API docs that had not yet made it to doc/docs/Python_User_Interface.md

### CODE REVIEW

File: doc/docs/Python_User_Interface.md [hunk 1]
```diff
@@ -107,8 +107,7 @@ def __init__(self,
              progress_interval=4,
              subpixel_tol=0.0001,
              subpixel_maxeval=100000,
-             loop_tile_base_db=0,
-             loop_tile_base_eh=0,
+             loop_tile_base=0,
```
Comment (oskooi) [inline doc/docs/Python_User_Interface.md:110]: This looks a regression due to the rebase.

File: doc/docs/Python_User_Interface.md [hunk 2]
```diff
@@ -287,13 +286,6 @@ Python. `Vector3` is a `meep` class.
   the minimum refractive index (usually 1), and in practice $S$ should be slightly
   smaller.
 
-+ **`loop_tile_base_db`, `loop_tile_base_eh` [`number`]** — To improve the [memory locality](https://en.wikipedia.org/wiki/Locality_of_reference)
-  of the field updates, Meep has an experimental feature to "tile" the loops over the Yee grid
-  voxels. The splitting of the update loops for step-curl and update-eh into tiles or subdomains
-  involves a recursive-bisection method in which the base case for the number of voxels is
-  specified using these two parameters, respectively. The default value is 0 or no tiling;
-  a typical nonzero value to try would be 10000.
-
```
Comment (stevengj) [inline doc/docs/Python_User_Interface.md:296]: Looks like a rebasing problem?

File: doc/docs/Python_User_Interface.md [hunk 3]
```diff
@@ -2311,11 +2296,21 @@ is a 1d array of `nfreq` dimensions.
...
-### Load and Dump Structure
+### Load and Dump Simulation State
 
-These functions dump the raw ε and μ data to disk and load it back for doing multiple simulations with the same materials but different sources etc. The only prerequisite is that the dump/load simulations have the same [chunks](Chunks_and_Symmetry.md) (i.e. the same grid, number of processors, symmetries, and PML). When using `split_chunks_evenly=False`, you must also dump the original chunk layout using `dump_chunk_layout` and load it into the new `Simulation` using the `chunk_layout` parameter. Currently only stores dispersive and non-dispersive $\varepsilon$ and $\mu$ but not nonlinearities. Note that loading data from a file in this way overwrites any `geometry` data passed to the `Simulation` constructor.
+The functions below add support for saving and restoring parts of the MEEP
+Simulation state.
+
+For all functions below, when dumping/loading state to/from a distributed filesystem
+(using say, parallel HDF5) and running in a MPI environment, setting
+`single_parallel_file=True` will result in all processes writing/reading to/from
```
Comment (stevengj) [inline doc/docs/Python_User_Interface.md:2306]:
```suggestion
`single_parallel_file=True` (the default) will result in all processes writing/reading to/from
```

File: doc/docs/Python_User_Interface.md [hunk 4]
```diff
@@ -2384,6 +2380,75 @@ sim2.load_minus_flux(...)
...
+</div>
+
+</div>
+
+#### Checkpoint / Restore.
+
+The above dump/load related functions can be used to implement a
+checkpoint/restore *like* feature for MEEP. The caveat of this feature is that
+it does *not* store all the state required to re-create the `Simulation` object
+itself. Instead in expects the user to create and setup the new `Simulation`
```
Comment (stevengj) [inline doc/docs/Python_User_Interface.md:2429]:
```suggestion
itself. Instead, it expects the user to create and set up the new `Simulation`
```

File: doc/docs/Python_User_Interface.md (hunk 2 above)
Comment (kkg4theweb) [reply doc/docs/Python_User_Interface.md:296]: Yeah. Please ignore this PR - Recreated this change in a  fresh branch
See https://github.com/NanoComp/meep/pull/1784

**Pull request closed without merging**